roofit/roofitcore/src/RooAbsPdf.cxx

```diff
@@ -2252,31 +2252,21 @@ RooDataSet *RooAbsPdf::generate(const RooArgSet &whatVars, const RooDataSet& pro
 
 Int_t* RooAbsPdf::randomizeProtoOrder(Int_t nProto, Int_t, Bool_t resampleProto) const
 {
-  // Make unsorted linked list of indeces
-  RooLinkedList l ;
-  Int_t i ;
-  for (i=0 ; i<nProto ; i++) {
-    l.Add(new RooInt(i)) ;
-  }
-
   // Make output list
   Int_t* lut = new Int_t[nProto] ;
 
   // Randomly samply input list into output list
   if (!resampleProto) {
     // In this mode, randomization is a strict reshuffle of the order
-    for (i=0 ; i<nProto ; i++) {
-      Int_t iran = RooRandom::integer(nProto-i) ;
-      RooInt* sample = (RooInt*) l.At(iran) ;
-      lut[i] = *sample ;
-      l.Remove(sample) ;
-      delete sample ;
+    std::iota(lut, lut + nProto, 0); // fill the vector with 0 to nProto - 1
+    // shuffle code taken from https://en.cppreference.com/w/cpp/algorithm/random_shuffle
+    // (std::random_shuffle was deprecated in C++17)
```

Comment:
this line because begs the question "why don't we use `std::shuffle` instead?" :smile: 

i guess it's because it's simpler to use `RooRandom::integer` than to set up an RNG from the STL,
but then maybe change this line with "for our usecase it's simpler to use RooRandom::integer than std::shuffle" or something like that

Reply:
Okay, I added some more lines of comments with my motivation for not using `std::shuffle`. I wanted to keep using `RooRandom::integer`, but `std::shuffle` expects a UniformRandomBitGenerator which RooRandom doesn't implement correctly.